src/core/qgsjsonutils.cpp

```diff
@@ -409,6 +409,12 @@ json QgsJsonUtils::jsonFromVariant( const QVariant &val )
     }
     j = jList;
   }
+  // to cover TextEdit Widget case
+  else if ( val.type() == QVariant::Type::String )
```

Comment:
Depending on the JSON implementation, a raw string is valid, so I'm hesitant to assume that "a raw string will contain JSON". I think that should go on another level (i.e. the text edit widget wrapper calling `parseJson`).

Reply:
Completed